File: algorithms/TwoPointers/python/125_validpalidrome.py
from typing import List

class Solution:
    def isPalindrome(self, s: str) -> bool:
        s_normalize = self.normalizeString(s)
        left, right = 0, len(s_normalize) - 1
        while left < right:
            if s_normalize[left] != s_normalize[right]:
                return False
            left += 1
            right -= 1
        return True

    def normalizeString(self, s: str) -> str:
        return ''.join(c.lower() for c in s if c.isalnum())

File: algorithms/TwoPointers/cpp/125_validpalidrome.cpp
#include<string>
#include<bits/stdc++.h>

using namespace std;

class Solution {
public:
    bool isPalindrome(string s) {
        string s_normalize = normalizeString(s);
        int left = 0, right = s_normalize.size() - 1;
        while (left < right) {
            if (s_normalize[left] != s_normalize[right]) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }

private:
    string normalizeString(const string &s) {
        string re;
        for(char c : s)
        {
            if(isalnum(static_cast<unsigned char>(c))){
                re += tolower(static_cast<unsigned char>(c));
            }
        }
        return re;
    }
};